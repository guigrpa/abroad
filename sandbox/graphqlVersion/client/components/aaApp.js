// @flow

/* eslint-env browser */

import React from 'react';
import { graphql } from 'react-relay';
import moment from 'moment';
import { Floats, Hints, Notifications, hintDefine, hintShow } from 'giu';
import type { KeyFilter } from '../gral/types';
import type { ViewerT } from '../../common/types';
import _t from '../../translate';
import { cookieGet, cookieSet } from '../gral/storage';
import { subscribe } from './helpers';
import updatedConfig from '../subscriptions/updatedConfig';
import updatedStats from '../subscriptions/updatedStats';
import parsedSrcFiles from '../subscriptions/parsedSrcFiles';
import updatedKey from '../subscriptions/updatedKey';
import createdTranslation from '../subscriptions/createdTranslation';
import updatedTranslation from '../subscriptions/updatedTranslation';
import Header from './abHeader';
import Translator from './adTranslator';
import Details from './afDetails';
import Settings from './aeSettings';
import QueryRendererWrapper from './uuQueryRendererWrapper';
import fetchLangBundle from './fetchLangBundle';

require('./aaApp.sass');

// Example MessageFormat message with plural, so that it appears in the screenshot:
// _t("someContext_{NUM, plural, one{1 hamburger} other{# hamburgers}}", { NUM: 1 })
// Example message with emoji, so that it appears in the screenshot:
// _t("someContext_Message with emoji: ðŸŽ‰")
// Example message with American and British English versions, so that it appears in the screenshot:
// _t("someContext_A tool for internationalization")
// Example of very long text
/* _t("someContext_Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin sit amet risus felis. Donec sed viverra urna. Praesent porttitor lorem id urna blandit, nec scelerisque risus varius. Nullam euismod suscipit scelerisque. Mauris placerat scelerisque nisi, vel rutrum purus dapibus et. Quisque mattis pulvinar eros in molestie. Suspendisse eleifend pretium nisi vel sagittis. In mollis diam sem, eget porta magna auctor nec. Maecenas rutrum eu magna sit amet feugiat. Nunc nec viverra dui. Suspendisse id lacus suscipit, volutpat sapien hendrerit, interdum risus. Mauris eu mollis lacus, ac posuere tortor. Integer quis pretium metus. Ut condimentum auctor elit, eu aliquet urna. Curabitur sit amet vulputate lorem, a lobortis risus.

Aliquam finibus lacinia iaculis. Nulla mattis lorem nec efficitur varius. Mauris tempus velit id auctor interdum. Sed facilisis, nunc id tincidunt sollicitudin, enim augue pulvinar ligula, vel consectetur lorem urna non velit. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Morbi eget auctor enim, sit amet euismod tellus. Integer vitae odio in ligula pretium interdum ac nec enim. Vestibulum tellus orci, hendrerit eu tortor quis, pretium pharetra massa. In quis quam venenatis, auctor eros nec, convallis felis. Donec mattis orci aliquam elit placerat, ac finibus turpis interdum. Proin lorem quam, dignissim vel quam ac, mollis imperdiet ipsum. Etiam iaculis elementum leo. Nulla eu lectus nec metus gravida congue. Pellentesque tempus mauris vel mauris lacinia, iaculis volutpat eros sodales. Cras eu leo ligula. Proin tortor ex, viverra efficitur risus nec, accumsan sollicitudin est.")
*/
// Example text in a sequence
// _t("someContext_Part 1 of the sequence")
// _t("someContext_Part 2 of the sequence")
// _t("someContext_Part 3 of the sequence")

// ==========================================
// Component declarations
// ==========================================
type Props = {
  // Unit testing
  _disableHints?: boolean,
  // Relay
  environment: Object,
  viewer: ViewerT,
};

const query = graphql`
  query aaAppQuery {
    viewer {
      id
      ...adTranslator_viewer
      config {
        ...aeSettings_config
      }
      keys(first: 100000) @connection(key: "App_viewer_keys") {
        edges {
          node {
            scope # for filtering
          }
        }
      }
    }
  }
`;

// ==========================================
// Component
// ==========================================
class App extends React.Component {
  props: Props;
  state: {
    selectedKeyId: ?string,
    fSettingsShown: boolean,
    lang: string,
    filter: KeyFilter,
    scope: ?string,
    quickFind: ?string,
  };

  isSubscribed = false;
  constructor() {
    super();
    this.state = {
      selectedKeyId: null,
      fSettingsShown: false,
      lang: cookieGet('lang', { defaultValue: 'en' }),
      filter: 'ALL',
      scope: null,
      quickFind: null,
    };
  }

  componentDidMount() {
    if (!this.props._disableHints) this.showHint();
    const { environment } = this.props;
    if (!environment) return;
    subscribe({ environment, subscriptionOptions: updatedConfig() });
    subscribe({ environment, subscriptionOptions: updatedStats() });
    subscribe({ environment, subscriptionOptions: parsedSrcFiles() });
    subscribe({ environment, subscriptionOptions: updatedKey() });
    subscribe({ environment, subscriptionOptions: createdTranslation() });
    subscribe({ environment, subscriptionOptions: updatedTranslation() });
  }

  // ------------------------------------------
  render() {
    const { viewer } = this.props;
    const scopes = this.getScopes();
    const { scope } = this.state;
    return (
      <div style={style.outer}>
        <Floats />
        <Notifications />
        <Hints />
        <Header
          lang={this.state.lang}
          onShowSettings={this.showSettings}
          filter={this.state.filter}
          changeFilter={this.changeFilter}
          scopes={scopes}
          scope={scope}
          changeScope={this.changeScope}
          quickFind={this.state.quickFind}
          changeQuickFind={this.changeQuickFind}
        />
        <Translator
          lang={this.state.lang}
          viewer={viewer}
          selectedKeyId={this.state.selectedKeyId}
          changeSelectedKey={this.changeSelectedKey}
          filter={this.state.filter}
          scope={scope}
          quickFind={this.state.quickFind}
        />
        {this.state.selectedKeyId && (
          <Details
            lang={this.state.lang}
            selectedKeyId={this.state.selectedKeyId}
          />
        )}
        {this.state.fSettingsShown && (
          <Settings
            lang={this.state.lang}
            config={viewer ? viewer.config : null}
            onChangeLang={this.onChangeLang}
            onClose={this.hideSettings}
          />
        )}
      </div>
    );
  }

  // ------------------------------------------
  changeSelectedKey = (selectedKeyId: ?string) => {
    this.setState({ selectedKeyId });
  };

  changeFilter = (filter: KeyFilter) => {
    this.setState({ filter });
  };

  changeScope = (scope: ?string) => {
    this.setState({ scope });
  };

  changeQuickFind = (quickFind: ?string) => {
    this.setState({ quickFind });
  };

  showSettings = () => {
    this.setState({ fSettingsShown: true });
  };
  hideSettings = () => {
    this.setState({ fSettingsShown: false });
  };

  onChangeLang = (lang: string) => {
    cookieSet('lang', lang);
    fetchLangBundle(lang, locales => {
      _t.setLocales(locales);
      moment.locale(lang);
      this.setState({ lang });
    });
  };

  // ------------------------------------------
  getScopes() {
    const { edges } = this.props.viewer.keys;
    const scopes = {};
    for (let i = 0; i < edges.length; i++) {
      const { scope } = edges[i].node;
      if (scope) scopes[scope] = true;
    }
    const scopesArr = Object.keys(scopes).sort();
    return scopesArr;
  }

  showHint(fForce: boolean = false) {
    const elements = () => {
      const out = [];
      const nodeSettings = document.getElementById('madyBtnSettings');
      if (nodeSettings) {
        const bcr = nodeSettings.getBoundingClientRect();
        const x = window.innerWidth - 150;
        out.push({
          type: 'LABEL',
          x,
          y: 70,
          align: 'right',
          children: _t('hint_Configure Mady'),
        });
        out.push({
          type: 'ARROW',
          from: { x, y: 70 },
          to: { x: (bcr.left + bcr.right) / 2, y: bcr.bottom },
          counterclockwise: true,
        });
      }
      const nodeAddLang = document.getElementById('madyBtnAddLang');
      if (nodeAddLang) {
        const bcr = nodeAddLang.getBoundingClientRect();
        const x = window.innerWidth - 80;
        out.push({
          type: 'LABEL',
          x,
          y: 140,
          align: 'right',
          children: _t('hint_Add language column'),
        });
        out.push({
          type: 'ARROW',
          from: { x, y: 140 },
          to: { x: (bcr.left + bcr.right) / 2, y: bcr.bottom },
          counterclockwise: true,
        });
      }
      const nodeFilterMenu = document.getElementById('madyMenuFilter');
      if (nodeFilterMenu) {
        const bcr = nodeFilterMenu.getBoundingClientRect();
        const x = 50;
        out.push({
          type: 'LABEL',
          x,
          y: 140,
          align: 'left',
          children: _t('hint_Filter relevant messages'),
        });
        out.push({
          type: 'ARROW',
          from: { x, y: 140 },
          to: { x: (bcr.left + bcr.right) / 2, y: bcr.bottom },
        });
      }
      return out;
    };
    const closeLabel = _t('hint_Enjoy translating!');
    hintDefine('main', { elements, closeLabel });
    hintShow('main', fForce);
  }
}

// ------------------------------------------
const style = {
  outer: {
    minHeight: '100%',
    padding: '0px 10px',
    display: 'flex',
    flexDirection: 'column',
    backgroundColor: 'white',
  },
};

// ==========================================
// Public API
// ==========================================
const Container = ({ relayEnvironment }: { relayEnvironment?: Object }) => (
  <QueryRendererWrapper
    environment={relayEnvironment}
    query={query}
    Component={App}
  />
);
export default Container;
export { query, App as _App };
