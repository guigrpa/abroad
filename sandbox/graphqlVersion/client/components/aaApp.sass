// ==============================================
// General mixins
// ==============================================
=boxSizing($value)
  -moz-box-sizing: #{$value}
  -webkit-box-sizing: #{$value}
  box-sizing: #{$value}

// ==============================================
// Fonts
// ==============================================
=systemFont
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif

// ==============================================
// Animations
// ==============================================
=animation($value)
  -webkit-animation: #{$value}
  -moz-animation: #{$value}
  -ms-animation: #{$value}
  -o-animation: #{$value}
  animation: #{$value}

=keyframes($value)
  @-webkit-keyframes #{$value}
  @-moz-keyframes #{$value}
  @-ms-keyframes #{$value}
  @-o-keyframes #{$value}
  @keyframes #{$value}

+keyframes('fadein {from {opacity: 0;} to {opacity: 1;}}')

.fadeIn
  +animation(fadein 0.4s)

// ==============================================
// Big things
// ==============================================
*,
*:before,
*:after
  +boxSizing(border-box)

html
  height: 100%
body
  padding: 0
  margin: 0
  height: 100%
  cursor: default
#mady-app
  +systemFont
  font-size: 12px
  height: 100%
input,
select
  +systemFont
  font-size: 12px

// ==============================================
// Small things
// ==============================================
.scrollbarMeasure
  position: fixed
  top: 0px
  left: 0px
  width: 100px
  height: 100px
  overflow: scroll
  opacity: 0.001
  top: -9999
