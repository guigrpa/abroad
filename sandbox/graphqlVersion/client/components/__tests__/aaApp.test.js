/* eslint-env jest */
/* eslint-disable global-require, import/newline-after-import */
import React from 'react';
import renderer from 'react-test-renderer';
import { _App as App } from '../aaApp';
import { VIEWER_WITH_NO_CONTENT } from './fixtures';
import $ from './jestQuery';

// https://github.com/facebook/react/issues/7386#issuecomment-238091398
jest.mock('react-dom');
jest.mock('../abHeader', () => require('./mockComponent')('Header'));
jest.mock('../adTranslator', () =>
  require('./mockComponent')('Translator', {
    description: ({ selectedKeyId }) => selectedKeyId || 'no selection',
  })
);
jest.mock('../aeSettings', () => require('./mockComponent')('Settings'));
jest.mock('../afDetails', () =>
  require('./mockComponent')('Details', {
    description: ({ selectedKeyId }) => selectedKeyId || 'no selection',
  })
);
jest.mock('../../gral/storage');
jest.mock('../fetchLangBundle');

describe('App', () => {
  it('01 no content', () => {
    const tree = renderer
      .create(<App viewer={VIEWER_WITH_NO_CONTENT} _disableHints />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('02 shows the settings when clicked', () => {
    const component = renderer.create(
      <App viewer={VIEWER_WITH_NO_CONTENT} _disableHints />
    );
    let tree = component.toJSON();

    const headerEl = $(tree, o => o.props.dataMockType === 'Header');
    expect(headerEl).not.toBeNull();
    headerEl.props.onShowSettings();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const settingsEl = $(tree, o => o.props.dataMockType === 'Settings');
    expect(settingsEl).not.toBeNull();
    settingsEl.props.onClose();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('03 changes the selected key when clicked and shows details', () => {
    const component = renderer.create(
      <App viewer={VIEWER_WITH_NO_CONTENT} _disableHints />
    );
    let tree = component.toJSON();

    const translatorEl = $(tree, o => o.props.dataMockType === 'Translator');
    expect(translatorEl).not.toBeNull();
    translatorEl.props.changeSelectedKey('SELECTED_KEY');
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('04 changes the application language', () => {
    const storage = require('../../gral/storage');
    const component = renderer.create(
      <App viewer={VIEWER_WITH_NO_CONTENT} _disableHints />
    );
    let tree = component.toJSON();

    const headerEl = $(tree, o => o.props.dataMockType === 'Header');
    expect(headerEl).not.toBeNull();
    headerEl.props.onShowSettings();
    tree = component.toJSON();

    const settingsEl = $(tree, o => o.props.dataMockType === 'Settings');
    expect(settingsEl).not.toBeNull();
    settingsEl.props.onChangeLang('ca');
    expect(storage.cookieGet('lang')).toBe('ca');
  });
});
