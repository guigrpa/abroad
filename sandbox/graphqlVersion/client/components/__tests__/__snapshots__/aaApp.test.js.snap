// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 01 no content 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "display": "flex",
      "flexDirection": "column",
      "minHeight": "100%",
      "padding": "0px 10px",
    }
  }
>
  <div
    className="giu-floats"
    style={
      Object {
        "height": 0,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 0,
      }
    }
  />
  <div
    className="giu-notifications"
    style={
      Object {
        "bottom": 20,
        "maxWidth": 350,
        "position": "fixed",
        "right": 20,
        "zIndex": 9000,
      }
    }
  />
  <div
    className="giu-hints"
  />
  <div
    changeFilter={[Function]}
    changeQuickFind={[Function]}
    changeScope={[Function]}
    dataMockType="Header"
    filter="ALL"
    lang="en"
    onShowSettings={[Function]}
    quickFind={null}
    scope={null}
    scopes={Array []}
  >
    Header
  </div>
  <div
    changeSelectedKey={[Function]}
    dataMockType="Translator"
    filter="ALL"
    lang="en"
    quickFind={null}
    scope={null}
    selectedKeyId={null}
    viewer={
      Object {
        "config": Object {
          "langs": Array [
            "es",
            "ca",
            "en",
          ],
        },
        "id": "me",
        "keys": Object {
          "edges": Array [],
        },
        "stats": Object {
          "numTotalKeys": 0,
          "numTranslations": Array [
            Object {
              "lang": "es",
              "value": 0,
            },
            Object {
              "lang": "ca",
              "value": 0,
            },
            Object {
              "lang": "en",
              "value": 0,
            },
          ],
          "numUsedKeys": 0,
        },
      }
    }
  >
    Translator (no selection)
  </div>
</div>
------------HTML PREVIEW---------------
<div style="background-color:white;display:flex;flex-direction:column;min-height:100%;padding:0px 10px;">
  <div
    class="giu-floats"
    style="height:0;left:0;position:absolute;top:0;width:0;"
  >
  </div>
  <div
    class="giu-notifications"
    style="bottom:20px;max-width:350px;position:fixed;right:20px;z-index:9000;"
  >
  </div>
  <div class="giu-hints"></div>
  <div
    dataMockType="Header"
    filter="ALL"
    lang="en"
  >
    Header
  </div>
  <div
    dataMockType="Translator"
    filter="ALL"
    lang="en"
  >
    Translator (no selection)
  </div>
</div>
`;

exports[`App 02 shows the settings when clicked 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "display": "flex",
      "flexDirection": "column",
      "minHeight": "100%",
      "padding": "0px 10px",
    }
  }
>
  <div
    className="giu-floats"
    style={
      Object {
        "height": 0,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 0,
      }
    }
  />
  <div
    className="giu-notifications"
    style={
      Object {
        "bottom": 20,
        "maxWidth": 350,
        "position": "fixed",
        "right": 20,
        "zIndex": 9000,
      }
    }
  />
  <div
    className="giu-hints"
  />
  <div
    changeFilter={[Function]}
    changeQuickFind={[Function]}
    changeScope={[Function]}
    dataMockType="Header"
    filter="ALL"
    lang="en"
    onShowSettings={[Function]}
    quickFind={null}
    scope={null}
    scopes={Array []}
  >
    Header
  </div>
  <div
    changeSelectedKey={[Function]}
    dataMockType="Translator"
    filter="ALL"
    lang="en"
    quickFind={null}
    scope={null}
    selectedKeyId={null}
    viewer={
      Object {
        "config": Object {
          "langs": Array [
            "es",
            "ca",
            "en",
          ],
        },
        "id": "me",
        "keys": Object {
          "edges": Array [],
        },
        "stats": Object {
          "numTotalKeys": 0,
          "numTranslations": Array [
            Object {
              "lang": "es",
              "value": 0,
            },
            Object {
              "lang": "ca",
              "value": 0,
            },
            Object {
              "lang": "en",
              "value": 0,
            },
          ],
          "numUsedKeys": 0,
        },
      }
    }
  >
    Translator (no selection)
  </div>
  <div
    config={
      Object {
        "langs": Array [
          "es",
          "ca",
          "en",
        ],
      }
    }
    dataMockType="Settings"
    lang="en"
    onChangeLang={[Function]}
    onClose={[Function]}
  >
    Settings
  </div>
</div>
------------HTML PREVIEW---------------
<div style="background-color:white;display:flex;flex-direction:column;min-height:100%;padding:0px 10px;">
  <div
    class="giu-floats"
    style="height:0;left:0;position:absolute;top:0;width:0;"
  >
  </div>
  <div
    class="giu-notifications"
    style="bottom:20px;max-width:350px;position:fixed;right:20px;z-index:9000;"
  >
  </div>
  <div class="giu-hints"></div>
  <div
    dataMockType="Header"
    filter="ALL"
    lang="en"
  >
    Header
  </div>
  <div
    dataMockType="Translator"
    filter="ALL"
    lang="en"
  >
    Translator (no selection)
  </div>
  <div
    dataMockType="Settings"
    lang="en"
  >
    Settings
  </div>
</div>
`;

exports[`App 02 shows the settings when clicked 2`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "display": "flex",
      "flexDirection": "column",
      "minHeight": "100%",
      "padding": "0px 10px",
    }
  }
>
  <div
    className="giu-floats"
    style={
      Object {
        "height": 0,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 0,
      }
    }
  />
  <div
    className="giu-notifications"
    style={
      Object {
        "bottom": 20,
        "maxWidth": 350,
        "position": "fixed",
        "right": 20,
        "zIndex": 9000,
      }
    }
  />
  <div
    className="giu-hints"
  />
  <div
    changeFilter={[Function]}
    changeQuickFind={[Function]}
    changeScope={[Function]}
    dataMockType="Header"
    filter="ALL"
    lang="en"
    onShowSettings={[Function]}
    quickFind={null}
    scope={null}
    scopes={Array []}
  >
    Header
  </div>
  <div
    changeSelectedKey={[Function]}
    dataMockType="Translator"
    filter="ALL"
    lang="en"
    quickFind={null}
    scope={null}
    selectedKeyId={null}
    viewer={
      Object {
        "config": Object {
          "langs": Array [
            "es",
            "ca",
            "en",
          ],
        },
        "id": "me",
        "keys": Object {
          "edges": Array [],
        },
        "stats": Object {
          "numTotalKeys": 0,
          "numTranslations": Array [
            Object {
              "lang": "es",
              "value": 0,
            },
            Object {
              "lang": "ca",
              "value": 0,
            },
            Object {
              "lang": "en",
              "value": 0,
            },
          ],
          "numUsedKeys": 0,
        },
      }
    }
  >
    Translator (no selection)
  </div>
</div>
------------HTML PREVIEW---------------
<div style="background-color:white;display:flex;flex-direction:column;min-height:100%;padding:0px 10px;">
  <div
    class="giu-floats"
    style="height:0;left:0;position:absolute;top:0;width:0;"
  >
  </div>
  <div
    class="giu-notifications"
    style="bottom:20px;max-width:350px;position:fixed;right:20px;z-index:9000;"
  >
  </div>
  <div class="giu-hints"></div>
  <div
    dataMockType="Header"
    filter="ALL"
    lang="en"
  >
    Header
  </div>
  <div
    dataMockType="Translator"
    filter="ALL"
    lang="en"
  >
    Translator (no selection)
  </div>
</div>
`;

exports[`App 03 changes the selected key when clicked and shows details 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "display": "flex",
      "flexDirection": "column",
      "minHeight": "100%",
      "padding": "0px 10px",
    }
  }
>
  <div
    className="giu-floats"
    style={
      Object {
        "height": 0,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 0,
      }
    }
  />
  <div
    className="giu-notifications"
    style={
      Object {
        "bottom": 20,
        "maxWidth": 350,
        "position": "fixed",
        "right": 20,
        "zIndex": 9000,
      }
    }
  />
  <div
    className="giu-hints"
  />
  <div
    changeFilter={[Function]}
    changeQuickFind={[Function]}
    changeScope={[Function]}
    dataMockType="Header"
    filter="ALL"
    lang="en"
    onShowSettings={[Function]}
    quickFind={null}
    scope={null}
    scopes={Array []}
  >
    Header
  </div>
  <div
    changeSelectedKey={[Function]}
    dataMockType="Translator"
    filter="ALL"
    lang="en"
    quickFind={null}
    scope={null}
    selectedKeyId="SELECTED_KEY"
    viewer={
      Object {
        "config": Object {
          "langs": Array [
            "es",
            "ca",
            "en",
          ],
        },
        "id": "me",
        "keys": Object {
          "edges": Array [],
        },
        "stats": Object {
          "numTotalKeys": 0,
          "numTranslations": Array [
            Object {
              "lang": "es",
              "value": 0,
            },
            Object {
              "lang": "ca",
              "value": 0,
            },
            Object {
              "lang": "en",
              "value": 0,
            },
          ],
          "numUsedKeys": 0,
        },
      }
    }
  >
    Translator (SELECTED_KEY)
  </div>
  <div
    dataMockType="Details"
    lang="en"
    selectedKeyId="SELECTED_KEY"
  >
    Details (SELECTED_KEY)
  </div>
</div>
------------HTML PREVIEW---------------
<div style="background-color:white;display:flex;flex-direction:column;min-height:100%;padding:0px 10px;">
  <div
    class="giu-floats"
    style="height:0;left:0;position:absolute;top:0;width:0;"
  >
  </div>
  <div
    class="giu-notifications"
    style="bottom:20px;max-width:350px;position:fixed;right:20px;z-index:9000;"
  >
  </div>
  <div class="giu-hints"></div>
  <div
    dataMockType="Header"
    filter="ALL"
    lang="en"
  >
    Header
  </div>
  <div
    dataMockType="Translator"
    filter="ALL"
    lang="en"
    selectedKeyId="SELECTED_KEY"
  >
    Translator (SELECTED_KEY)
  </div>
  <div
    dataMockType="Details"
    lang="en"
    selectedKeyId="SELECTED_KEY"
  >
    Details (SELECTED_KEY)
  </div>
</div>
`;
